src/platform/datapath_raw_xdp_win.c

```diff
@@ -371,6 +382,38 @@ CxPlatDpRawInterfaceUninitialize(
     #pragma warning(pop)
 }
 
+QUIC_STATUS
+GetTxOffloadConfig(
+    _In_ CXPLAT_QUEUE* Queue,
+    _Out_ uint32_t* TxChecksumOffload
+    )
+{
+    QUIC_STATUS Status;
+    XDP_CHECKSUM_CONFIGURATION ChecksumConfig;
+    uint32_t OptionLength = sizeof(ChecksumConfig);
+
+    Status =
+        XskGetSockopt(
+            Queue->TxXsk, XSK_SOCKOPT_TX_OFFLOAD_CURRENT_CONFIG_CHECKSUM, &ChecksumConfig,
+            &OptionLength);
+    if (QUIC_FAILED(Status) ||
+        OptionLength < XDP_SIZEOF_CHECKSUM_CONFIGURATION_REVISION_1 ||
+        ChecksumConfig.Header.Revision != XDP_CHECKSUM_CONFIGURATION_REVISION_1 ||
+        ChecksumConfig.Header.Size < XDP_SIZEOF_CHECKSUM_CONFIGURATION_REVISION_1) {
+        *TxChecksumOffload = FALSE;
+        if (QUIC_SUCCEEDED(Status)) {
+            Status = QUIC_STATUS_NOT_SUPPORTED;
+        }
+        goto Exit;
+    }
+
+    *TxChecksumOffload = ChecksumConfig.Enabled;
```

Comment:
Should we log (informational?) a success result too?

Reply:
Yeah. I'm not sure how to (cheaply) add an informational log, though.

Reply:
```c
QuicTraceLogInfo(
    XdpInterfaceGetOffload,
    "[data][%p] Shut down (complete)",
    Socket);
```
Something like the above, you can use printf syntax.